ttnn/CMakeLists.txt

```diff
@@ -37,20 +37,27 @@ set(TTNN_SRCS
     ${CMAKE_CURRENT_SOURCE_DIR}/cpp/ttnn/operations/data_movement/pad/device/pad_op.cpp
 )
 
-add_library(ttnn_lib OBJECT ${TTNN_SRCS})
-target_compile_options(ttnn_lib PUBLIC -MP -Wno-int-to-pointer-cast -fno-var-tracking)
-target_link_libraries(ttnn_lib
-    PUBLIC compiler_flags metal_header_directories metal_common_libs
-)
-target_include_directories(ttnn_lib PUBLIC
+### Setup TTNN_LIB as an Object library
+# Save variables to reuse between ttnn_lib and ttnn
+# Have to manually propagate because ttnn_lib is an OBJECT library
```

Comment:
Really? Object libraries don't propagate this stuff through their public interface? Didn't know that!

Reply:
Yes. They do not propagate up from their own dependencies. Very annoying.